Gist/EngineWeb/EngineWeb.swift

```diff
@@ -41,9 +41,12 @@ public class EngineWeb: NSObject {
         webView.backgroundColor = UIColor.clear
         webView.scrollView.backgroundColor = UIColor.clear
 
-        let contentController = self.webView.configuration.userContentController
-        contentController.add(self, name: "gist")
-
+        let js = "window.parent.postMessage = function(message) {webkit.messageHandlers.gist.postMessage(message)}"
+        let messageHandlerScript = WKUserScript(source: js, injectionTime: .atDocumentEnd, forMainFrameOnly: false)
+        
+        self.webView.configuration.userContentController.add(self, name: "gist")
+        self.webView.configuration.userContentController.addUserScript(messageHandlerScript)
```

Comment:
I am not so familiar with Gist code but what I understand is that this change logically does a similar action as it would on `didFinish` delegate method, just that it won't wait for the delegate method to be called and will inject the script at the end of the code. Though it looks good to me but I am curious to know how this would solve the issue we're facing i.e rendering the messages.

Reply:
Yeah, i'm sure this is the safest way to make sure it's injected at the best time. However, I also tried adding the script manually to the renderer page and we're still seeing the issue.